python/examples/core/harness.py

```diff
@@ -74,6 +76,81 @@ def to_data_frame(self) -> pandas.DataFrame:
     """Return a data frame containing the aggregated data."""
     return self.data
 
+  def plot(self, path: os.path, data_key: str, data_label: str):
+    """Plot the measurements."""
+    config_key = "runtime_problem_sizes_dict"
+    keys_to_iter = [k for k in self.config_keys if k is not config_key]
+    config_to_iter = self.data[keys_to_iter].drop_duplicates()
+    for _, row in config_to_iter.iterrows():
+      # Compute the plot file name by filtering non alpha numberic characters.
+      file_name = "plot"
+      for k, v in row.to_dict().items():
+        alnum_name = ''.join([c for c in v if c.isalnum()])
+        file_name += str.format(f"_{k}_{alnum_name}")
+      file_name += ".pdf"
+      file_name = os.path.join(path, file_name)
+      # Obtain the data elements to plot.
+      filtered = self.data
+      for k, v in row.to_dict().items():
+        filtered = filtered[filtered[k] == v]
+      # Plot the selected data.
+      plt = seaborn.violinplot(x=config_key, y=data_key, data=filtered)
+      keys, new_labels = self._compress_problem_sizes_label(
+          [text.get_text() for text in plt.xaxis.get_ticklabels()])
+      plt.set(xticklabels=new_labels)
+      plt.set(xlabel=str.format(
+          f"problem sizes [{','.join(keys)}]"), ylabel=data_label)
+      plt.tick_params(axis='x', rotation=20)
+      fig = plt.get_figure()
+      fig.tight_layout()
+      fig.savefig(file_name)
+
+  def _compress_problem_sizes_label(self,
+              labels: Sequence[str]) -> (Sequence[str], Sequence[str]):
+    """Compress the problem size labels.
+
+    Identify the dimensions that take different values and compress the
+    labels to contain only the values for these varying dimensions.
```

Comment:
would be nice to add a line to explain why this is useful

Reply:
done